src/test/java/org/broadinstitute/hellbender/utils/config/ConfigUnitTest.java

```diff
@@ -0,0 +1,277 @@
+
+    // ================================================================================
+    // Helper Methods:
+    // ================================================================================
+
+    void listAndStoreConfigToStdOut(Accessible config) throws IOException {
+
+        config.list(System.out);
+        System.out.println();
+
+        config.store(System.out, "");
+        System.out.println();
+
+        config.storeToXML(System.out, "");
+        System.out.println();
+    }
+
+    void validateAndClearProperty(String key, Object value) {
+
+        String other = null;
+
+        if ( value != null ) {
+
+            StringBuilder sb = new StringBuilder();
+            sb.append(value);
+
+            other = sb.toString();
+        }
+
+        Assert.assertEquals(
+                System.getProperty(key),
+                other
+        );
+
+        System.clearProperty(key);
+    }
+
+    // ================================================================================
+    // Data Providers:
+    // ================================================================================
+
+    @DataProvider
+    Object[][] createArgsAndConfigFileOptions() {
+        return new Object[][] {
+                {
+                    new ArrayList<>(Arrays.asList(new String[] {"main","--zonfigurati","DUMMY_FILE","END"})),
+                    "--config",
+                    null,
+                    new ArrayList<>(Arrays.asList(new String[] {"main","--zonfigurati","DUMMY_FILE","END"})),
+                },
+                {
+                    new ArrayList<>(Arrays.asList(new String[] {"main","--config","DUMMY_FILE","END"})),
+                    "--config",
+                    "DUMMY_FILE",
+                    new ArrayList<>(Arrays.asList(new String[] {"main","END"})),
+                },
+                {
+                    new ArrayList<>(Arrays.asList(new String[] {"main","END","--config","DUMMY_FILE"})),
+                    "--config",
+                    "DUMMY_FILE",
+                    new ArrayList<>(Arrays.asList(new String[] {"main","END"})),
+                },
+        };
+    }
+
+    @DataProvider
+    Object[][] createArgsAndConfigFileOptionsBadInput() {
+        return new Object[][] {
+                {
+                        new ArrayList<>(Arrays.asList(new String[] {"main", "testArg", "--config"})),
+                        "--config",
+                },
+                {
+                        new ArrayList<>(Arrays.asList(new String[] {"main","--config"})),
+                        "--config",
+                },
+                {
+                        new ArrayList<>(Arrays.asList(new String[] {"--config"})),
+                        "--config",
+                },
+        };
+    }
+
+    // ================================================================================
+    // Tests:
```

Comment:
These comment blocks are fairly unusual in our code.  They can definitely help make things clearer in large classes, but they're not part of our standard practice.